stdlib/public/SDK/Dispatch/Time.swift

```diff
@@ -114,6 +114,30 @@ extension DispatchWallTime {
   }
 }
 
+
+// Returns m1 * m2, clamped to the range [Int64.min, Int64.max].
+// Because of the way this function is used, we can always assume
+// that m2 > 0.
+private func clampedInt64Product(_ m1: Int64, _ m2: Int64) -> Int64 {
+	assert(m2 > 0, "multiplier must be positive")
+	if m1 == 0 { return Int64(0) }
+	if m1 > Int64.max/m2 { return Int64.max }
+	if m1 < Int64.min/m2 { return Int64.min }
+	return m1 * m2;
```

Comment:
You could use `multiplyWithOverflow` for this instead…

Reply:
Do you mean multipliedReportingOverflow()? I didn't know about that :-)

Reply:
Regarding the change overall, the general rule in libdispatch is that operations on dispatch_time_t that overflow are clamped to DISPATCH_TIME_FOREVER. The intent of this change is to make the same true in Swift. For the change to DispatchTimeInterval, you are right that it might seem odd that DispatchTimeInterval.seconds(Int.max) == .milliseconds(Int.max), but DispatchTimeInterval is used as an offset to form a DispatchTime or DispatchWallTime and the result of using either of these values is going to be clamped to Dispatch[Wall]Time.distantFuture. Currently, you can't say "DispatchTime.now() + DispatchTimeInterval.seconds(Int.max)" because it traps. Of course Int.max is a pathological example - in reality, there have been cases where large values have been used that have caused traps and this had to be worked around by checking the value before passing it to dispatch and using DispatchTime.distantFuture instead. I probably should add some comments to DispatchTimeInterval that explain this behavior.